packages/material-ui/src/Theme/MaterialUIContextProps.ts

```diff
@@ -1,69 +1,69 @@
-/** Use require for loading these libraries in case they are not available in order to perform a useful fallback
- */
-let mui = {};
-try {
-  mui = require('@material-ui/core');
-} catch (err) {
-  // purposely a no-op
-}
-
-// @ts-ignore What we are doing here isn't really good Typescript, but it works
-const {
-  Box,
-  Button,
-  Checkbox,
-  Divider,
-  FormControl,
-  FormControlLabel,
-  FormGroup,
-  FormHelperText,
-  FormLabel,
-  Grid,
-  IconButton,
-  Input,
-  InputLabel,
-  List,
-  ListItem,
-  ListItemIcon,
-  ListItemText,
-  MenuItem,
-  Paper,
-  Radio,
-  RadioGroup,
-  Slider,
-  SvgIcon,
-  TextField,
-  Typography,
-} = mui;
+import { ComponentType } from 'react';
 
 export default interface MaterialUIContextProps {
-  Box?: Box;
-  Button?: Button;
-  Checkbox?: Checkbox;
-  Divider?: Divider;
-  Grid?: Grid;
-  FormControl?: FormControl;
-  FormControlLabel?: FormControlLabel;
-  FormGroup?: FormGroup;
-  FormHelperText?: FormHelperText;
-  FormLabel?: FormLabel;
-  IconButton?: IconButton;
-  Input?: Input;
-  InputLabel?: InputLabel;
-  List?: List;
-  ListItem?: ListItem;
-  ListItemIcon?: ListItemIcon;
-  ListItemText?: ListItemText;
-  MenuItem?: MenuItem;
-  Paper?: Paper;
-  Radio?: Radio;
-  RadioGroup?: RadioGroup;
-  Slider?: Slider;
-  TextField?: TextField;
-  Typography?: Typography;
-  AddIcon?: SvgIcon;
-  ArrowDownwardIcon?: SvgIcon;
-  ArrowUpwardIcon?: SvgIcon;
-  ErrorIcon?: SvgIcon;
-  RemoveIcon?: SvgIcon;
+  AddIcon: ComponentType<any>;
+  ArrowDownwardIcon: ComponentType<any>;
+  ArrowUpwardIcon: ComponentType<any>;
+  Box: ComponentType<any>;
+  Button: ComponentType<any>;
+  Checkbox: ComponentType<any>;
+  Divider: ComponentType<any>;
+  ErrorIcon: ComponentType<any>;
+  FormControl: ComponentType<any>;
+  FormControlLabel: ComponentType<any>;
+  FormGroup: ComponentType<any>;
+  FormHelperText: ComponentType<any>;
+  FormLabel: ComponentType<any>;
```

Comment:
Is it OK that we're removing the types now?

Reply:
I'm worried about that as well...

Reply:
I believe those types are only useful at time of `react-json-schemna-form` development and should not impact consumers. Typescript was fine with the switchover to a generic ComponentType for my local `react-json-schema-form` development. We could import the component props for each component type and pass in to the generic (instead of `any`) but I don't think it's necessary.